commands/lib/search.go

```diff
@@ -18,59 +18,77 @@
 package lib
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"strings"
 
 	"github.com/arduino/arduino-cli/arduino/libraries/librariesindex"
-	"github.com/arduino/arduino-cli/cli"
-	"github.com/arduino/arduino-cli/common/formatter"
-	"github.com/arduino/arduino-cli/common/formatter/output"
-	"github.com/sirupsen/logrus"
-	"github.com/spf13/cobra"
+	"github.com/arduino/arduino-cli/commands"
+	"github.com/arduino/arduino-cli/rpc"
 )
 
-func initSearchCommand() *cobra.Command {
-	searchCommand := &cobra.Command{
-		Use:     "search [LIBRARY_NAME]",
-		Short:   "Searchs for one or more libraries data.",
-		Long:    "Search for one or more libraries data (case insensitive search).",
-		Example: "  " + cli.AppName + " lib search audio",
-		Args:    cobra.ArbitraryArgs,
-		Run:     runSearchCommand,
-	}
-	searchCommand.Flags().BoolVar(&searchFlags.names, "names", false, "Show library names only.")
-	return searchCommand
-}
-
-var searchFlags struct {
-	names bool // if true outputs lib names only.
-}
+func LibrarySearch(ctx context.Context, req *rpc.LibrarySearchReq) (*rpc.LibrarySearchResp, error) {
 
-func runSearchCommand(cmd *cobra.Command, args []string) {
-	logrus.Info("Executing `arduino lib search`")
-	query := strings.ToLower(strings.Join(args, " "))
+	lm := commands.GetLibraryManager(req)
+	if lm == nil {
+		return nil, errors.New("invalid instance")
+	}
 
-	lm := cli.InitLibraryManager(cli.Config)
+	res := []*rpc.SearchLibraryOutput{}
 
-	res := output.LibSearchResults{
-		Libraries: []*librariesindex.Library{},
-	}
 	for _, lib := range lm.Index.Libraries {
-		if strings.Contains(strings.ToLower(lib.Name), query) {
-			res.Libraries = append(res.Libraries, lib)
+		if strings.Contains(strings.ToLower(lib.Name), req.GetQuery()) {
+			release := map[string]*rpc.LibraryRelease{}
+			for str, rel := range lib.Releases {
+				release[str] = GetLibraryParameters(rel)
+			}
+			latest := GetLibraryParameters(lib.Latest)
+
+			searchedlib := &rpc.SearchLibraryOutput{
+				Name:     lib.Name,
+				Releases: release,
+				Latest:   latest,
+			}
+			res = append(res, searchedlib)
 		}
 	}
 
-	if searchFlags.names {
-		for _, lib := range res.Libraries {
-			formatter.Print(lib.Name)
+	if req.GetNames() {
+		restmp := []*rpc.SearchLibraryOutput{}
+		for _, lib := range res {
+			searchedlib := &rpc.SearchLibraryOutput{
+				Name: lib.Name,
+			}
+			restmp = append(restmp, searchedlib)
 		}
+		res = restmp
 	} else {
-		if len(res.Libraries) == 0 {
-			formatter.Print(fmt.Sprintf("No library found matching `%s` search query", query))
-		} else {
-			formatter.Print(res)
+		if len(res) == 0 {
+			return nil, fmt.Errorf("%s library not found", req.GetQuery())
```

Comment:
I think we should not error but just return an empty list.

Reply:
i'm agree and done